server/server.go

```diff
@@ -525,15 +529,75 @@ func (s *Server) loadIndex(id string) error {
 	return nil
 }
 
+const repositoryIDSize = sha256.Size
+const RepositoryVersion = 1
+
+func (s *Server) createConfig() (err error) {
+	s.Config.ChunkerPolynomial, err = chunker.RandomPolynomial()
+	if err != nil {
+		return err
+	}
+
+	newID := make([]byte, repositoryIDSize)
+	_, err = io.ReadFull(rand.Reader, newID)
+	if err != nil {
+		return err
+	}
+
+	s.Config.ID = hex.EncodeToString(newID)
+	s.Config.Version = RepositoryVersion
+
+	debug.Log("Server.createConfig", "New config: %#v", s.Config)
+
+	_, err = s.SaveJSONUnpacked(backend.Config, s.Config)
+	return err
+}
+
+func (s *Server) loadConfig(cfg *Config) error {
+	err := s.LoadJSONUnpacked(backend.Config, nil, cfg)
+	if err != nil {
+		return err
+	}
+
+	if !cfg.ChunkerPolynomial.Irreducible() {
```

Comment:
I've just added another benchmark, running this irreducibility check costs 300ms, this would be run for every file, so I think it's too expensive to run it on every `chunker.New()`. Thoughts?